code-challenges.test.js

```diff
@@ -14,39 +14,169 @@
+// inside the method create if else statements to change each vowel for its associated number in the test variables. Include the upper case letters to handle all occasions.
+// return the new array with .join() to turn the array of charaters back into a string.
+
+const codeCreator = (string) => {
+  let array = string.split("")
+
+  newArray = array.map((letter) => {
+    if (letter === "a" || letter === "A") {
+      return "4"
+    } else if (letter === "e" || letter === "E") {
+      return "3"
+    } else if (letter === "i" || letter === "I") {
+      return "1"
+    } else if (letter === "o" || letter === "O") {
+      return "0"
+    } else {
+      return letter
+    }
+  })
+  return newArray.join("")
+}
+
+// PASS  ./code-challenges.test.js
+// PASS  ./jest.test.js
+
+// Test Suites: 2 passed, 2 total
+// Tests:       2 passed, 2 total
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
-
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+describe("wordChecker", () => {
+  const fruitArray = [
+    "Mango",
+    "Cherry",
+    "Apricot",
+    "Blueberry",
+    "Peach",
+    "Kiwi",
+  ]
+
+  const letterA = "a"
+  // Expected output: ["Mango", "Apricot", "Peach"]
+  const letterE = "e"
+  // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+  it("Function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter", () => {
+    expect(wordChecker(fruitArray, letterA)).toEqual([
+      "Mango",
+      "Apricot",
+      "Peach",
+    ])
+    expect(wordChecker(fruitArray, letterE)).toEqual([
+      "Cherry",
+      "Blueberry",
+      "Peach",
+    ])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+// Puesdo Code
+// Create the function wordChecker to match the test.
+// Create a new variable to store the updated array.
+// inside the varible use .filter for the original array.
+// Make the value for .fitler 'word' to help with the flow.
+// inside of .filter create if else statements to see if the 'word' has the parameter string.
+// This can be done by using the .includes method
+// if evalutaed as true return the word.
+// Check for uppercase by doing the same thing in an else if statemnet but add .toUpperCase to alter the string and check for capitals.
+// Also return word for this.
+// after this add on an else that returns nothing.
+// return the newArray.
+
+const wordChecker = (array, string) => {
+  const newArray = array.filter((word) => {
+    if (word.includes(string)) {
+      return word
+    } else if (word.includes(string.toUpperCase())) {
```

Comment:
Well done! Just a note: the `.filter()` method has a built-in `if` statement, which means you don't have to add one! You can think of it like this: any logic that comes after the arrow syntax within the `.filter()` method is essentially living in an `if` statement by default.